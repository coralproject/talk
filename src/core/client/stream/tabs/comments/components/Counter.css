.root {
  composes: subTabCounter from "talk-ui/shared/typography.css";

  background-color: var(--palette-grey-dark);
  color: var(--palette-text-light);
  border-radius: 1px;
  padding: 2px 3px 3px 3px;
}
