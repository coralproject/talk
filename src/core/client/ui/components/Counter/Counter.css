.root {
  composes: subTabCounter from "talk-ui/shared/typography.css";
  border-radius: 1px;
  padding: 2px 3px 3px 3px;
  color: inherit;
}

.colorPrimary {
  background-color: var(--palette-primary-main);
}

.colorGrey {
  background-color: var(--palette-grey-dark);
}

.colorInherit {
  background-color: currentColor;
}

.text {
  color: var(--palette-text-light);
}
