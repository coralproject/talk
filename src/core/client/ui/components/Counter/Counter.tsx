import cn from "classnames";
import React, { StatelessComponent } from "react";

import { withStyles } from "talk-ui/hocs";

import styles from "./Counter.css";

interface Props {
  className?: string;
  /**
   * Override or extend the styles applied to the component.
   */
  classes: typeof styles;
  /**
   * The color of the component. It supports those theme colors that make sense for this component.
   */
  color?: "inherit" | "grey" | "primary";
}

const Counter: StatelessComponent<Props> = ({
  classes,
  color,
  className,
  children,
  ...rest
}) => {
  const rootClassName = cn(
    classes.root,
    {
      [classes.colorPrimary]: color === "primary",
      [classes.colorGrey]: color === "grey",
      [classes.colorInherit]: color === "inherit",
    },
    className
  );
  return (
    <span className={rootClassName} {...rest}>
      <span className={classes.text}>{children}</span>
    </span>
  );
};

Counter.defaultProps = {
  color: "inherit",
};

const enhanced = withStyles(styles)(Counter);
export default enhanced;
