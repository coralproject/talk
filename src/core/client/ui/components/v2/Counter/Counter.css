$counterBadgeForegroundColor: var(--v2-colors-pure-white);
$counterBadgeBackgroundDefaultColor: var(--v2-colors-grey-500);
$counterBadgeBackgroundEmphasisColor: var(--v2-colors-grey-500);
$counterBadgeBackgroundAlertColor: var(--v2-colors-grey-500);

.root {
  font-size: var(--v2-font-size-2);
  font-weight: var(--v2-font-weight-primary-bold);
  font-family: var(--v2-font-family-primary);
  line-height: var(--v2-line-height-reset);
  letter-spacing: calc(0.2em / 14);
  border-radius: var(--round-corners);
  padding: 1px 3px 1px 3px;
  color: inherit;
}

.sizeSmall {
  font-size: var(--v2-font-size-1);
  padding: 2px 3px;
  line-height: var(--v2-line-height-reset);
}

.text {
  color: $counterBadgeForegroundColor;
}

.colorInherit {
  background-color: currentColor;
}

.colorPrimary {
  background-color: var(--palette-primary-main);
}

.colorGrey {
  background-color: var(--palette-grey-lightest);
  box-shadow: inset 0px 0px 0px 1px var(--palette-grey-dark);
  & .text {
    color: var(--palette-grey-dark);
  }
}

.colorDark {
  background-color: var(--palette-grey-dark);
}

.colorError {
  background-color: var(--palette-error-darkest);
}

.colorEmphasis {
  background-color: $counterBadgeBackgroundEmphasisColor;
}

.colorDefault {
  background-color: $counterBadgeBackgroundDefaultColor;
}

.colorAlert {
  background-color: $counterBadgeBackgroundAlertColor;
}
