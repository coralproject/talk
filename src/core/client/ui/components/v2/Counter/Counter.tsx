import cn from "classnames";
import React, { FunctionComponent } from "react";

import { withStyles } from "coral-ui/hocs";

import styles from "./Counter.css";

interface Props {
  className?: string;
  /**
   * Override or extend the styles applied to the component.
   */
  classes: typeof styles;
  /**
   * The color of the component. It supports those theme colors that make sense for this component.
   */
  color?:
    | "inherit"
    | "dark"
    | "grey"
    | "primary"
    | "error"
    | "default"
    | "emphasis"
    | "alert";

  size?: "sm" | "md";
}

const Counter: FunctionComponent<Props> = ({
  classes,
  color,
  className,
  children,
  size,
  ...rest
}) => {
  const rootClassName = cn(
    classes.root,
    {
      [classes.colorPrimary]: color === "primary",
      [classes.colorDark]: color === "dark",
      [classes.colorGrey]: color === "grey",
      [classes.colorInherit]: color === "inherit",
      [classes.colorError]: color === "error",
      [classes.colorDefault]: color === "default",
      [classes.colorEmphasis]: color === "emphasis",
      [classes.colorAlert]: color === "alert",
      [classes.sizeSmall]: size === "sm",
    },
    className
  );
  return (
    <span className={rootClassName} {...rest}>
      <span className={classes.text}>{children}</span>
    </span>
  );
};

Counter.defaultProps = {
  color: "inherit",
};

const enhanced = withStyles(styles)(Counter);
export default enhanced;
