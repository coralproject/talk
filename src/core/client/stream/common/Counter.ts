import cn from "classnames";
import { withProps } from "recompose";

import CLASSES from "coral-stream/classes";
import { Counter } from "coral-ui/components";
import { PropTypesOf } from "coral-ui/types";

export default withProps<
  PropTypesOf<typeof Counter>,
  PropTypesOf<typeof Counter>
>(({ className }) => ({
  className: cn(className, CLASSES.counter),
}))(Counter);
