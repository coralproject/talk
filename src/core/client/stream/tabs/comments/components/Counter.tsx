import cn from "classnames";
import React, { StatelessComponent } from "react";

import * as styles from "./Counter.css";

interface Props {
  className?: string;
}

const Counter: StatelessComponent<Props> = props => {
  return (
    <span className={cn(styles.root, props.className)}>{props.children}</span>
  );
};

export default Counter;
