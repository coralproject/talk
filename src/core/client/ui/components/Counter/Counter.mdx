---
name: Counter
menu: UI Kit
---

import { Playground, PropsTable } from "docz";

import Flex from "../Flex";
import Counter from "./Counter";

# Counter

## Basic usage

<Playground>
  <Flex alignItems="center">
    <span>Comments</span> <Counter>23</Counter>
  </Flex>
</Playground>

<Playground>
  <Flex itemGutter>
    <Counter size="sm">23</Counter>
    <Counter size="sm" color="primary">
      15
    </Counter>
    <Counter size="sm" color="grey">
      10
    </Counter>
    <Counter size="sm" color="dark">
      20
    </Counter>
  </Flex>
</Playground>
