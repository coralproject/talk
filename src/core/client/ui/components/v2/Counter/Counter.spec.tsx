import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import Counter from "./Counter";

it("renders correctly", () => {
  const renderer = createRenderer();
  renderer.render(<Counter>20</Counter>);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
