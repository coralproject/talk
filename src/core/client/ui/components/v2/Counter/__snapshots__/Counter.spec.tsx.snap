// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Counter
  classes={
    Object {
      "colorAlert": "Counter-colorAlert",
      "colorDark": "Counter-colorDark",
      "colorDefault": "Counter-colorDefault",
      "colorEmphasis": "Counter-colorEmphasis",
      "colorError": "Counter-colorError",
      "colorGrey": "Counter-colorGrey",
      "colorInherit": "Counter-colorInherit",
      "colorPrimary": "Counter-colorPrimary",
      "root": "Counter-root",
      "sizeSmall": "Counter-sizeSmall",
      "text": "Counter-text",
    }
  }
  color="inherit"
>
  20
</Counter>
`;
